app/models/grouping.rb

```diff
@@ -578,9 +578,14 @@ def create_grouping_repository_folder
         if revision.path_exists?(assignment_folder)
           return true
         else
-          txn = self.group.repo.get_transaction('markus')
+          txn = repo.get_transaction('markus')
           txn.add_path(assignment_folder)
-          return self.group.repo.commit(txn)
+          # If it is a git repo, we have to add a keep file so
+          # the assignment folder gets committed.
+          if repo.instance_of?(Repository::GitRepository)
```

Comment:
Can this be done in the git_repository.rb file instead of here?  It would be better to keep repo specific code out of the models.  Maybe in make_directory?